### PR TITLE
Bug 1343681 - Fix tagsChanged logic

### PR DESCRIPTION
**Pull request by soltysh**
@smarterclayton you gonna love this one :stuck_out_tongue_winking_eye:

### CODE REVIEW

File: pkg/image/api/helper.go [hunk 1]
```diff
@@ -615,7 +615,7 @@ func tagsChanged(new, old []TagEvent) (changed bool, deleted bool) {
 	case len(old) == 0:
 		return true, false
 	default:
-		return new[0] == old[0], false
+		return new[0] != old[0], false
```
Comment (smarterclayton) [inline pkg/image/api/helper.go]: AWESOME

**Comment by smarterclayton** [comment]
LGTM [merge] thanks

**Comment by soltysh** [comment]
It's failing unit tests though :( digging into it.

**Comment by soltysh** [comment]
It looks like `old` is being modified in `tagsChanged` in `pkg/image/registry/imagestram/strategy.go`, looking into it.

**Comment by soltysh** [comment]
@smarterclayton I've removed your merge tag, because I'd want your 2nd look. Esp. are you ok with the change in `strategy_test.go` which copies the `existingTagHistory` map (in `old` stream), so that it's not modified along with the new stream.

**Pull request merged**